### PR TITLE
Migrate DoB and pronouns to user model

### LINKED ISSUE #503: migrate volunteer DoB and pronouns to user schema
- update all types and code across backend and frontend
- for forms, make sure we are now submitting DoB and pronouns to the create user queries
- As part of this ticket, please ensure that that the edit account pages `edit-account` are all updated and functional when querying for data

### PR DESCRIPTION
**Pull request by eevanwong**
## Ticket link

Closes #503

## Implementation description
* Changed prisma schema, changed corresponding types in backend and frontend, as well as to graphql schemas.
* 

## Steps to test
1. Use any query with users and they will now have pronouns and date of birth associated with them.

As a general note: I noticed that, while the prisma schema originally made pronouns essential and dateOfBirth nullable, the typing in types didnt reflect that (sometimes both were nullable). How important is it that the type `string` vs `string | null`?

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/graphql/types/userType.ts [hunk 1]
```diff
@@ -112,7 +120,7 @@ const userType = gql`
     password: String!
     hireDate: Date!
     dateOfBirth: Date
-    pronouns: String
+    pronouns: String!
```
Comment (MatoPlus) [inline backend/graphql/types/userType.ts:123]: Like you pointed out in the PR desc, was this originally defaulted to "" or something similar in the service implementation for the resolver?

File: backend/graphql/types/userType.ts [hunk 2]
```diff
@@ -129,7 +137,7 @@ const userType = gql`
     emergencyContactEmail: String
     hireDate: Date!
     dateOfBirth: Date
-    pronouns: String
+    pronouns: String!
```
Comment (MatoPlus) [inline backend/graphql/types/userType.ts:140]: idem

File: backend/graphql/types/userType.ts (hunk 1 above)
Comment (MatoPlus) [reply backend/graphql/types/userType.ts:123]: https://github.com/uwblueprint/sistering/pull/541/files#diff-9fb1b1d1450d36b06c56517448127a1728819a646db02b29716efa1f89c8df97L751-L752

found it. Hmm I think there was some miscommunication during dev, pronouns should be required according to the account from: https://www.figma.com/file/tOK6AjkaRZ1NBXo191D64C/Sistering-%2F-User-Authentication-Process?node-id=518%3A6949

File: backend/services/implementations/userService.ts [hunk 3]
```diff
@@ -856,6 +870,8 @@ class UserService implements IUserService {
                 emergencyContactName: volunteerUser.emergencyContactName,
                 emergencyContactPhone: volunteerUser.emergencyContactPhone,
                 emergencyContactEmail: volunteerUser.emergencyContactEmail,
+                pronouns: volunteerUser.pronouns ?? "",
```
Comment (MatoPlus) [inline backend/services/implementations/userService.ts:873]: Could we clear up this and make pronouns required as well?
```suggestion
                pronouns: volunteerUser.pronouns ??,
```

File: backend/services/implementations/userService.ts [hunk 4]
```diff
@@ -979,14 +993,12 @@ class UserService implements IUserService {
               emergencyContactName: deletedVolunteerUser.emergencyContactName,
               emergencyContactPhone: deletedVolunteerUser.emergencyContactPhone,
               emergencyContactEmail: deletedVolunteerUser.emergencyContactEmail,
+              pronouns: deletedVolunteerUser.pronouns ?? "",
```
Comment (MatoPlus) [inline backend/services/implementations/userService.ts:996]: idem, should not have fallback

**Review by MatoPlus — CHANGES_REQUESTED** [review]
Made some suggestions, looks like we also need to update https://github.com/uwblueprint/sistering/pull/541/files# since pronouns is now a required field to create users.

File: backend/services/implementations/userService.ts (hunk 3 above)
Comment (eevanwong) [reply backend/services/implementations/userService.ts:873]: The suggested change isnt valid, I need an expression there, do you mean just remove the ?? operator?

File: backend/services/implementations/userService.ts (hunk 4 above)
Comment (eevanwong) [reply backend/services/implementations/userService.ts:996]: Ok, will remove, I assume thats what you meant in the initial suggestion as well.

File: backend/services/implementations/userService.ts (hunk 3 above)
Comment (MatoPlus) [reply backend/services/implementations/userService.ts]: > do you mean just remove the ?? operator?
Yes, that was what I was trying to suggest.

**Review by MatoPlus — COMMENTED** [review]
can we query the new `pronouns` and `dateOfBirth` field in the `EMPLOYEE_BY_ID` query under the Edit Account page? The updated query should look something like:
```gql
const EMPLOYEE_BY_ID = gql`
  query EmployeeUserById($id: ID!) {
    employeeUserById(id: $id) {
      id
      firstName
      lastName
      email
      dateOfBirth
      pronouns
      phoneNumber
      emergencyContactPhone
      languages
      branches {
        id
        name
      }
    }
  }
`;

```

**Review by MatoPlus — APPROVED** [review]
LGTM with quick suggestion.

**Comment by MatoPlus** [comment]
Looks like we'll have to reset the staging db for this. Please let me know when merging this.

**Review by MatoPlus — APPROVED** [review]
Thanks for getting this done @eevanwong!

**Pull request merged**